pandas/core/dtypes/cast.py

```diff
@@ -2096,6 +2096,9 @@ def maybe_cast_to_integer_array(
         )
         return casted
 
+    if lib.infer_dtype(casted) == "integer":
```

Comment:
i think this _always_ holds.  The condition we're interested in is whether the casting was lossy.

Reply:
Agreed, but this test currently passes the extra test added.

Reply:
do the added tests pass w/o this change?

Reply:
Yes they do.

Reply:
Right, because this condition _always_ holds.

Reply:
But before adding this check the testcase in the issue was not passing.

Reply:
> But before adding this check the testcase in the issue was not passing.

That's a reason to add _some_ check for this, but this particular check is not the right check.  ATM this is equivalent to (but slower than) `if True:`